### ISSUE TITLE
Consistent damaged Amor plane-stress reduction and energy derivatives

### ISSUE THREAD

**Issue by allamaprabhuani** [CEMS-Lab/PhAST#6]
## Reproduction

Verified against public main `95397f78ce7136ad08e8fd9e52f96cf4024a8c77`
using Python 3.10.18 and PyTorch 2.8.0 on CPU in float64. This example uses
only the public package and a single synthetic triangle.

```python
import torch
from phast.mesh import FEMMesh
from phast.material import Material
from phast.fem_operators import FEMOperators

nodes = torch.tensor([[0., 0.], [1., 0.], [0., 1.]], dtype=torch.float64)
mesh = FEMMesh.from_tensors(
    nodes, torch.tensor([[0, 1, 2]]), device="cpu", dtype=torch.float64
)
material = Material(
    E=6000., nu=.22, rho=1., Gc=1., l0=.2, eta_residual=1e-5,
    energy_split="amor", pf_model="AT2", plane_stress=True
)
op = FEMOperators(mesh, material)
q = torch.tensor([-.001, -.001, 0.], dtype=torch.float64, requires_grad=True)
d = q.new_tensor(.5)
strain = tuple(q.unbind())
g = material.degradation(d)
stress = torch.stack(op.compute_stress_amor(*strain, g))
plus = op.compute_psi_plus(None, strain=strain)
energy = g * plus + op._psi_minus_for_energy(strain, d, plus)
print(stress.detach().tolist())
print(torch.autograd.grad(energy, q)[0].tolist())
```

Observed stress is `[-5.76925, -5.76925, 0]`. The gradient of the reported
energy is `[-4.684448224852072, -4.684448224852072, 0]`. The third strain
coordinate is engineering shear, so the corresponding derivative is shear
stress. Stress units follow the supplied Young's modulus.

## Physical Interpretation

The current stress and positive-energy paths recover the elastic out-of-plane
strain as `ezz = -nu/(1-nu) * (exx+eyy)`. Evaluating the damaged three-dimensional
Amor stress at that recovery gives `sigma_zz = -3.846115384615384` for this
state. This quantity is reconstructed from the inspected formula. It is not
an out-of-plane state returned by the public API.

Eliminating `ezz` from the damaged three-dimensional energy at fixed damage
gives normal stress `-3.000073199590083` and zero out-of-plane stress to
roundoff. The compressive recovery depends on degradation, which also affects
the damage-driving and implicit differentiation rules.

The stress JVP can agree with finite differences of the implemented stress
map while this physical inconsistency remains. A separate bounded check gave
relative agreement of approximately `1.01e-11` for one smooth direction.

## Acceptance

- Establish a consistent plane-stress reduction with zero out-of-plane stress.
- Check returned stress against the reduced-energy derivative and verify its tangent.
- Keep damage driving and implicit backward consistent with the same reduction.
- Include intact, tensile, plane-strain and damaged-compressive controls.
- Check branch membership for the actual finite-difference perturbations.
- Expose or test the actual recovered constitutive state for future acceptance.

A stress-only change is insufficient. This material-point counterexample
does not measure trajectory error. Historical results should retain their
original source identities while a consistent correction is developed.

[Public stress implementation](https://github.com/CEMS-Lab/PhAST/blob/95397f78ce7136ad08e8fd9e52f96cf4024a8c77/src/phast/core/fem_operators.py)
